app/assets/javascripts/client/consensus/ConsensusService.js

```diff
@@ -0,0 +1,153 @@
+(function() {
+  'use strict';
+  angular.module('PDRClient')
+      .service('ConsensusService', ConsensusService);
+
+  ConsensusService.$inject = [
+    '$stateParams',
+    '$location',
+    '$q',
+    'Consensus',
+    'AnalysisConsensus',
+    'AnalysisResponse',
+    'Score',
+    'AnalysisConsensusScore',
+    'ProductEntry',
+    'DataEntry'
+  ];
+
+  function ConsensusService($stateParams, $location, $q, Consensus, AnalysisConsensus, AnalysisResponse, Score, AnalysisConsensusScore, ProductEntry, DataEntry) {
+    var service = this;
+
+    service.setContext = function(context) {
+      service.context = context;
+      return service;
+    };
+
+    service.extractId = function() {
+      return $stateParams.assessment_id || $stateParams.analysis_id;
+    };
+
+    service.redirectToIndex = function () {
+      if (service.context === "assessment") {
+        $location.path("/assessments");
```

Comment:
I wonder if we should be using `$state.transitionTo` here instead, although if it fetches the same results then I don't a reason to change it now.